testutils/Makefile

```diff
@@ -0,0 +1,74 @@
+GO111MODULE = on
+export GO111MODULE
+GOFLAGS ?= -mod=vendor
+export GOFLAGS
+GOPROXY ?=
+export GOPROXY
+
+# ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
+ENVTEST_K8S_VERSION = 1.26
+
+PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
+ENVTEST = go run ${PROJECT_DIR}/vendor/sigs.k8s.io/controller-runtime/tools/setup-envtest
+GOLANGCI_LINT = go run ${PROJECT_DIR}/vendor/github.com/golangci/golangci-lint/cmd/golangci-lint
+
+NO_DOCKER ?= 0
+
+ifeq ($(shell command -v podman > /dev/null 2>&1 ; echo $$? ), 0)
+	ENGINE=podman
+else ifeq ($(shell command -v docker > /dev/null 2>&1 ; echo $$? ), 0)
+	ENGINE=docker
+else
+	NO_DOCKER=1
+endif
+
+USE_DOCKER ?= 0
+ifeq ($(USE_DOCKER), 1)
+	ENGINE=docker
+endif
+
+ifeq ($(NO_DOCKER), 1)
+  DOCKER_CMD =
+else
+  DOCKER_CMD := $(ENGINE) run \
+	  --rm \
+	  -v "$(PWD)/..":/go/src/github.com/openshift/cluster-api-actuator-pkg:Z \
+	  -w /go/src/github.com/openshift/cluster-api-actuator-pkg/testutils \
+	  -e "GO111MODULE=$(GO111MODULE)" \
+	  -e "GOFLAGS=$(GOFLAGS)" \
+	  -e "GOPROXY=$(GOPROXY)" \
+	  registry.ci.openshift.org/openshift/release:golang-1.19
+  IMAGE_BUILD_CMD = $(ENGINE) build
+endif
+
+.PHONY: all
+all: check
+
+.PHONY: check
+check: lint fmt goimports
+
+.PHONY: vendor
+vendor:
+	go mod tidy
+	go mod vendor
+	go mod verify
+
+.PHONY: lint
+lint: ## Go lint your code
+	( GOLANGCI_LINT_CACHE=$(PROJECT_DIR)/.cache $(GOLANGCI_LINT) run --config ../.golangci.yaml --timeout 10m )
+
+.PHONY: fmt
+fmt: ## Go fmt your code
+	$(DOCKER_CMD) hack/go-fmt.sh .
+
+.PHONY: goimports
+goimports: ## run goimports
+	$(DOCKER_CMD) hack/goimports.sh .
```

Comment:
Are these not covered by GolangCI Lint? My understanding was they are both run as part of its suite

Reply:
You are correct. I have kept them because, they were being executed by CI. There are currently two targets that run on this repo. (excluding E2Es). `check` that on main branch runs fmt and vet, and goimports. The golanCI lint has been added recently and is not being executed by CI. If you want, we can remove these and only keep golangci lint and unit.

Reply:
If that's the case, then yep, lets keep them for now but I think we should migrate to golangci-lint as the single source moving forward, that can be a follow up PR though